### PR TITLE
Add support for keyvault

### PR DESCRIPTION
**Pull request by javiercn**

### CODE REVIEW

File: src/Microsoft.AspNetCore.Identity.Service.KeyVault/KeyVaultSigningCredentialsSourceOptions.cs [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+using System.Net.Http;
+
+namespace Microsoft.AspNetCore.Identity.Service.KeyVault
+{
+    public class KeyVaultSigningCredentialsSourceOptions
+    {
+        public string ClientId { get; set; }
+        public string ClientSecret { get; set; }
+        public string CertificateThumbprint { get; set; }
```
Comment (pakrym) [inline src/Microsoft.AspNetCore.Identity.Service.KeyVault/KeyVaultSigningCredentialsSourceOptions.cs]: Do we want to put thumbprint here or maybe certificate object?

Comment (javiercn) [inline src/Microsoft.AspNetCore.Identity.Service.KeyVault/KeyVaultSigningCredentialsSourceOptions.cs]: @pakrym I would keep the config simple (and bindable) as much as possible, and have the component load the certificate. This version only supports ClientSecret for now

Comment (pakrym) [inline src/Microsoft.AspNetCore.Identity.Service.KeyVault/KeyVaultSigningCredentialsSourceOptions.cs]: Let's remove `CertificateThumbprint` until we start supporting it.

File: src/Microsoft.AspNetCore.Identity.Service.KeyVault/IdentityServiceBuilderExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+            builder.Services.Configure(configure);
+            builder.Services.TryAddSingleton<ISigningCredentialsSource, KeyVaultSigningCredentialSource>();
+
+            return builder;
+        }
+
+        private class DefaultSetup : ConfigureOptions<KeyVaultSigningCredentialsSourceOptions>
+        {
+            public DefaultSetup(IConfiguration configuration)
```
Comment (pakrym) [inline src/Microsoft.AspNetCore.Identity.Service.KeyVault/IdentityServiceBuilderExtensions.cs]: I think @danroth27 had an idea how to conditionally enable default options binding. We may want to use it here.

**Comment by danroth27** [comment]
Fixes #1211

**Pull request merged**